Matchmaker/CassandraMatchRequestRepository.cs

```diff
@@ -1,47 +1,69 @@
 using Cassandra.Mapping;
+using Cassandra;
 
 public class CassandraMatchRequestRepository : IMatchRequestRepository
 {
     private IMapper db;
+    private ConsistencyLevel _consistencyLevel;
     private Random random = new();
 
-    public CassandraMatchRequestRepository(IMapper cassandraMapper)
+    public CassandraMatchRequestRepository(int port, ConsistencyLevel consistencyLvl)
     {
-        db = cassandraMapper;
+        var cluster = Cluster.Builder()
+                     .AddContactPoint("127.0.0.1")
+                     .WithPort(port)
+                     .Build();
+
+        var session = cluster.Connect("matchmaker");
+
+        db = new Mapper(session);
+        _consistencyLevel = consistencyLvl;
     }
 
     public IEnumerable<MatchRequest> GetByGameTypeAndRegion(GameType gameType, Region region, int limit)
     {
         var randomGuid = Guid.NewGuid();
-        var firstPart = db.Fetch<MatchRequest>(
+
+        Cql cql = new Cql(
             "WHERE gametype=? AND region=? AND playerid>=? LIMIT ?",
             (int)gameType,
             (int)region,
             randomGuid,
-            limit
-        ).ToList();
-        var secondPart = ((firstPart.Count() < limit) ? db.Fetch<MatchRequest>(
+            limit).WithOptions(x => x.SetConsistencyLevel(_consistencyLevel));
+
+        var firstPart = db.Fetch<MatchRequest>(cql).ToList();
+
+        cql = new Cql(
             "WHERE gametype=? AND region=? AND playerid<? LIMIT ?",
             (int)gameType,
             (int)region,
             randomGuid,
-            limit - firstPart.Count()
-        ) : Enumerable.Empty<MatchRequest>()).ToList();
+            limit - firstPart.Count()).WithOptions(x => x.SetConsistencyLevel(_consistencyLevel));
+
+        var secondPart = ((firstPart.Count() < limit) ? db.Fetch<MatchRequest>(cql) : Enumerable.Empty<MatchRequest>()).ToList();
+        
         return firstPart.Union(secondPart);
     }
 
     public IEnumerable<MatchRequest> GetByPriority(int priority, int limit)
     {
-        return db.Fetch<MatchRequest>($"WHERE priority=? LIMIT ?", priority, limit);
+        Cql cql = new Cql($"WHERE priority=? LIMIT ?", priority, limit)
+            .WithOptions(x => x.SetConsistencyLevel(_consistencyLevel));
+        return db.Fetch<MatchRequest>(cql);
     }
 
-    public void RemoveByPlayerId(Guid playerId)
+    public void RemoveByPlayerId(Guid playerId, Region region, GameType gameType)
```

Comment:
Tu myślałem o usuwaniu wszystkich rekordów z danym `playerId`, ale chyba ostatecznie i tak nie mamy przypadków, żeby jeden gracz miał wiele różnych requestów, więc możemy tak to zrobić.

Reply:
Tutaj dałem region i typ, do BD zwracał błąd